### PR TITLE
unification - currency `type`

### PR DESCRIPTION
**Pull request by ttodua**
top exchanges had mostly `crypto` and `fiat` currencies as unified. so, (beside `info` exchange specific props) we have to unify those types (some exchanges return  i.e. `'CryptoCurrency'` or alike strings) however, we shouldn't return those raw strings to user, but unified type (like we have it for markets: `spot/swap/etc`).  at this moment, while testing all exchanges against `type` field, I have seen 3 common types:
```
- crypto
- fiat
- other
```
so, I've updated all exchanges to be consistent with those returned values. (note, `other` is not same as `undefined`, but it means that currency is somewhat other type, some exchanges call it strings like `'unknown', 'other', 'alternative', etc... , so it is not "undefined" type, but "other" is the common name. if someone on the earth is looking for exchange-specific value, they can use `info` for non-unified raw data)

### CODE REVIEW

**Comment by ttodua** [comment]
@carlosmiei if you'd have few mins to review this

File: ts/src/ndax.ts [hunk 1]
```diff
@@ -362,7 +362,12 @@ export default class ndax extends Exchange {
             const currency = response[i];
             const id = this.safeString (currency, 'ProductId');
             const name = this.safeString (currency, 'ProductFullName');
-            const type = this.safeString (currency, 'ProductType');
+            const ProductType = this.safeString (currency, 'ProductType');
+            let type = ProductType === 'NationalCurrency' ? 'fiat' : 'crypto';
```
Comment (carlosmiei) [inline ts/src/ndax.ts:366]: @ttodua can you add () to these conditions to avoid issues transpiling it?

Comment (ttodua) [inline ts/src/ndax.ts:366]:
```suggestion
            let type = (ProductType === 'NationalCurrency') ? 'fiat' : 'crypto';
```

File: ts/src/kucoin.ts [hunk 2]
```diff
@@ -1126,10 +1128,13 @@ export default class kucoin extends Exchange {
                     },
                 };
             }
+            // kucoin has determined 'fiat' currencies with below logic
+            const isFiat = rawPrecision === '2' && chainsLength === 0;
```
Comment (ttodua) [inline ts/src/kucoin.ts:1132]:
```suggestion
            const isFiat = (rawPrecision === '2') && (chainsLength === 0);
```

File: ts/src/ndax.ts (hunk 1 above)
Comment (ttodua) [reply ts/src/ndax.ts]: uh oh, forgot them. now done & pushed.

**Comment by carlosmiei** [comment]
LGTM ✅

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**